// @strapi/admin/build/20.9e5a98b6.chunk.js
(self.webpackChunk_strapi_admin=self.webpackChunk_strapi_admin||[]).push([[20],{12001:(X,ae,A)=>{"use strict";X.exports=A(27901)},27901:function(X,ae,A){var V=A(98399);(function(Y,q){X.exports=q(A(32735),A(19615),A(60672))})(this,function(Y,q,ne){return function(r){var l={};function e(o){if(l[o])return l[o].exports;var a=l[o]={i:o,l:!1,exports:{}};return r[o].call(a.exports,a,a.exports,e),a.l=!0,a.exports}return e.m=r,e.c=l,e.d=function(o,a,f){e.o(o,a)||Object.defineProperty(o,a,{enumerable:!0,get:f})},e.r=function(o){typeof Symbol!="undefined"&&Symbol.toStringTag&&Object.defineProperty(o,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(o,"__esModule",{value:!0})},e.t=function(o,a){if(1&a&&(o=e(o)),8&a||4&a&&typeof o=="object"&&o&&o.__esModule)return o;var f=Object.create(null);if(e.r(f),Object.defineProperty(f,"default",{enumerable:!0,value:o}),2&a&&typeof o!="string")for(var u in o)e.d(f,u,function(d){return o[d]}.bind(null,u));return f},e.n=function(o){var a=o&&o.__esModule?function(){return o.default}:function(){return o};return e.d(a,"a",a),a},e.o=function(o,a){return Object.prototype.hasOwnProperty.call(o,a)},e.p="",e(e.s=99)}([function(r,l,e){r.exports=e(19)()},function(r,l){r.exports=Y},function(r,l){r.exports=q},function(r,l){r.exports=function(e,o){return o||(o=e.slice(0)),Object.freeze(Object.defineProperties(e,{raw:{value:Object.freeze(o)}}))},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l,e){var o=e(24);r.exports=function(a,f){if(a==null)return{};var u,d,p=o(a,f);if(Object.getOwnPropertySymbols){var b=Object.getOwnPropertySymbols(a);for(d=0;d<b.length;d++)u=b[d],f.indexOf(u)>=0||Object.prototype.propertyIsEnumerable.call(a,u)&&(p[u]=a[u])}return p},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){function e(){return r.exports=e=Object.assign||function(o){for(var a=1;a<arguments.length;a++){var f=arguments[a];for(var u in f)Object.prototype.hasOwnProperty.call(f,u)&&(o[u]=f[u])}return o},r.exports.default=r.exports,r.exports.__esModule=!0,e.apply(this,arguments)}r.exports=e,r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l,e){"use strict";e.r(l),e.d(l,"Box",function(){return w});var o,a=e(3),f=e.n(a),u=e(2),d=e.n(u),p=e(7),b=e(1),g=e.n(b),O=e(0),i=e.n(O),h=function(n){return g.a.createElement("div",n)},v={background:void 0,borderColor:void 0,color:void 0,hiddenS:!1,hiddenXS:!1,padding:void 0,paddingTop:void 0,paddingRight:void 0,paddingBottom:void 0,paddingLeft:void 0,hasRadius:!1,shadow:void 0,children:null,shrink:void 0,grow:void 0,basis:void 0,flex:void 0,_hover:function(){}},x={_hover:i.a.func,background:i.a.string,basis:i.a.oneOfType([i.a.string,i.a.string]),borderColor:i.a.string,children:i.a.oneOfType([i.a.node,i.a.string]),color:i.a.string,flex:i.a.oneOfType([i.a.string,i.a.string]),grow:i.a.oneOfType([i.a.string,i.a.string]),hasRadius:i.a.bool,hiddenS:i.a.bool,hiddenXS:i.a.bool,padding:i.a.oneOfType([i.a.number,i.a.arrayOf(i.a.number)]),paddingBottom:i.a.oneOfType([i.a.number,i.a.arrayOf(i.a.number)]),paddingLeft:i.a.oneOfType([i.a.number,i.a.arrayOf(i.a.number)]),paddingRight:i.a.oneOfType([i.a.number,i.a.arrayOf(i.a.number)]),paddingTop:i.a.oneOfType([i.a.number,i.a.arrayOf(i.a.number)]),shadow:i.a.string,shrink:i.a.oneOfType([i.a.string,i.a.string])};h.defaultProps=v,h.propTypes=x;var T={color:!0},w=d.a.div.withConfig({shouldForwardProp:function(n,t){return!T[n]&&t(n)}})(o||(o=f()([`
  // Font
  font-size: `,`;

  // Colors
  background: `,`;
  color: `,`;

  // Spaces
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`
  `,`

  // Responsive hiding
  `,`
  `,`
  

  // Borders
  border-radius: `,`;
  border-style: `,`;
  border-width: `,`;
  border-color: `,`;
  border: `,`;

  // Shadows
  box-shadow: `,`;

  // Handlers
  pointer-events: `,`;
  &:hover {
    `,`
  }

  // Display
  display: `,`;

  // Position
  position: `,`;
  left: `,`;
  right: `,`;
  top: `,`;
  bottom: `,`;
  z-index: `,`;
  overflow: `,`;
  cursor: `,`;

  // Size
  width: `,`;
  max-width: `,`;
  min-width: `,`;
  height: `,`;
  max-height: `,`;
  min-height: `,`;

  // Animation
  transition: `,`;
  transform: `,`;
  animation: `,`;

  //Flexbox children props
  flex-shrink: `,`;
  flex-grow: `,`;
  flex-basis: `,`;
  flex: `,`;

  // Text
  text-align: `,`;
  text-transform: `,`;
  line-height: `,`;

  // Cursor
  cursor: `,`;
`])),function(n){var t=n.fontSize;return n.theme.fontSizes[t]||t},function(n){var t=n.theme,c=n.background;return t.colors[c]},function(n){var t=n.theme,c=n.color;return t.colors[c]},function(n){var t=n.theme,c=n.padding;return Object(p.a)("padding",c,t)},function(n){var t=n.theme,c=n.paddingTop;return Object(p.a)("padding-top",c,t)},function(n){var t=n.theme,c=n.paddingRight;return Object(p.a)("padding-right",c,t)},function(n){var t=n.theme,c=n.paddingBottom;return Object(p.a)("padding-bottom",c,t)},function(n){var t=n.theme,c=n.paddingLeft;return Object(p.a)("padding-left",c,t)},function(n){var t=n.theme,c=n.marginLeft;return Object(p.a)("margin-left",c,t)},function(n){var t=n.theme,c=n.marginRight;return Object(p.a)("margin-right",c,t)},function(n){var t=n.theme,c=n.marginTop;return Object(p.a)("margin-top",c,t)},function(n){var t=n.theme,c=n.marginBottom;return Object(p.a)("margin-bottom",c,t)},function(n){var t=n.theme;return n.hiddenS?"".concat(t.mediaQueries.tablet," { display: none; }"):void 0},function(n){var t=n.theme;return n.hiddenXS?"".concat(t.mediaQueries.mobile," { display: none; }"):void 0},function(n){var t=n.theme,c=n.hasRadius,S=n.borderRadius;return c?t.borderRadius:S},function(n){return n.borderStyle},function(n){return n.borderWidth},function(n){var t=n.borderColor;return n.theme.colors[t]},function(n){var t=n.theme,c=n.borderColor,S=n.borderStyle,C=n.borderWidth;if(c&&!S&&!C)return"1px solid ".concat(t.colors[c])},function(n){var t=n.theme,c=n.shadow;return t.shadows[c]},function(n){return n.pointerEvents},function(n){var t=n._hover,c=n.theme;return t?t(c):void 0},function(n){return n.display},function(n){return n.position},function(n){var t=n.left;return n.theme.spaces[t]||t},function(n){var t=n.right;return n.theme.spaces[t]||t},function(n){var t=n.top;return n.theme.spaces[t]||t},function(n){var t=n.bottom;return n.theme.spaces[t]||t},function(n){return n.zIndex},function(n){return n.overflow},function(n){return n.cursor},function(n){var t=n.width;return n.theme.spaces[t]||t},function(n){var t=n.maxWidth;return n.theme.spaces[t]||t},function(n){var t=n.minWidth;return n.theme.spaces[t]||t},function(n){var t=n.height;return n.theme.spaces[t]||t},function(n){var t=n.maxHeight;return n.theme.spaces[t]||t},function(n){var t=n.minHeight;return n.theme.spaces[t]||t},function(n){return n.transition},function(n){return n.transform},function(n){return n.animation},function(n){return n.shrink},function(n){return n.grow},function(n){return n.basis},function(n){return n.flex},function(n){return n.textAlign},function(n){return n.textTransform},function(n){return n.lineHeight},function(n){return n.cursor});w.defaultProps=v,w.propTypes=x},function(r,l,e){"use strict";var o=e(10),a=e.n(o),f=e(13),u=e.n(f);l.a=function(d,p,b){var g=p;if(Array.isArray(p)||u()(p)!=="object"||(g=[p==null?void 0:p.desktop,p==null?void 0:p.tablet,p==null?void 0:p.mobile]),g!==void 0){if(Array.isArray(g)){var O=g,i=a()(O,3),h=i[0],v=i[1],x=i[2],T="".concat(d,": ").concat(b.spaces[h],";");return v!==void 0&&(T+="".concat(b.mediaQueries.tablet,`{
          `).concat(d,": ").concat(b.spaces[v],`;
        }`)),x!==void 0&&(T+="".concat(b.mediaQueries.mobile,`{
          `).concat(d,": ").concat(b.spaces[x],`;
        }`)),T}var w=b.spaces[g]||g;return"".concat(d,": ").concat(w,";")}}},function(r,l,e){"use strict";e.r(l),e.d(l,"Typography",function(){return w});var o,a=e(3),f=e.n(a),u=e(2),d=e.n(u),p=["alpha","beta","delta","epsilon","omega","pi","sigma"],b=e(1),g=e.n(b),O=e(0),i=e.n(O),h=function(n){return g.a.createElement("div",n)},v={ellipsis:!1,fontWeight:void 0,fontSize:void 0,lineHeight:void 0,textColor:void 0,textTransform:void 0,variant:"omega"},x={ellipsis:i.a.bool,fontSize:i.a.oneOfType([i.a.number,i.a.string]),fontWeight:i.a.string,lineHeight:i.a.oneOfType([i.a.number,i.a.string]),textColor:i.a.string,textTransform:i.a.string,variant:i.a.oneOf(p)};h.defaultProps=v,h.propTypes=x;var T={fontSize:!0,fontWeight:!0},w=d.a.span.withConfig({shouldForwardProp:function(n,t){return!T[n]&&t(n)}})(o||(o=f()([`
  font-weight: `,`;
  font-size: `,`;
  line-height: `,`;
  color: `,`;
  text-transform: `,`;
  `,`
  `,`
`])),function(n){var t=n.theme,c=n.fontWeight;return t.fontWeights[c]},function(n){var t=n.theme,c=n.fontSize;return t.fontSizes[c]},function(n){var t=n.theme,c=n.lineHeight;return t.lineHeights[c]},function(n){var t=n.theme,c=n.textColor;return t.colors[c||"neutral800"]},function(n){return n.textTransform},function(n){return n.ellipsis&&`
    display: block;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  `},function(n){var t=n.variant,c=n.theme;switch(t){case"alpha":return`
        font-weight: `.concat(c.fontWeights.bold,`;
        font-size: `).concat(c.fontSizes[5],`;
        line-height: `).concat(c.lineHeights[2],`;
      `);case"beta":return`
        font-weight: `.concat(c.fontWeights.bold,`;
        font-size: `).concat(c.fontSizes[4],`;
        line-height: `).concat(c.lineHeights[1],`;
      `);case"delta":return`
        font-weight: `.concat(c.fontWeights.semiBold,`;
        font-size: `).concat(c.fontSizes[3],`;
        line-height: `).concat(c.lineHeights[2],`;
      `);case"epsilon":return`
        font-size: `.concat(c.fontSizes[3],`;
        line-height: `).concat(c.lineHeights[6],`;
      `);case"omega":return`
        font-size: `.concat(c.fontSizes[2],`;
        line-height: `).concat(c.lineHeights[4],`;
      `);case"pi":return`
        font-size: `.concat(c.fontSizes[1],`;
        line-height: `).concat(c.lineHeights[3],`;
      `);case"sigma":return`
        font-weight: `.concat(c.fontWeights.bold,`;
        font-size: `).concat(c.fontSizes[0],`;
        line-height: `).concat(c.lineHeights[5],`;
        text-transform: uppercase;
      `);default:return`
        font-size: `.concat(c.fontSizes[2],`;
      `)}});w.defaultProps=v,w.propTypes=x},function(r,l,e){"use strict";e.r(l),e.d(l,"Flex",function(){return n});var o,a=e(3),f=e.n(a),u=e(2),d=e.n(u),p=e(6),b=e(7),g=e(1),O=e.n(g),i=e(0),h=e.n(i),v=function(t){return O.a.createElement("div",t)},x={alignItems:"center",basis:void 0,direction:"row",gap:void 0,inline:!1,justifyContent:void 0,reverse:!1,wrap:void 0},T={alignItems:h.a.string,basis:h.a.oneOfType([h.a.string,h.a.number]),direction:h.a.string,gap:h.a.oneOfType([h.a.shape({desktop:h.a.number,mobile:h.a.number,tablet:h.a.number}),h.a.number,h.a.arrayOf(h.a.number),h.a.string]),inline:h.a.bool,justifyContent:h.a.string,reverse:h.a.bool,wrap:h.a.string};v.defaultProps=x,v.propTypes=T;var w={direction:!0},n=d()(p.Box).withConfig({shouldForwardProp:function(t,c){return!w[t]&&c(t)}})(o||(o=f()([`
  align-items: `,`;
  display: `,`;
  flex-direction: `,`;
  flex-wrap: `,`;
  `,`};
  justify-content: `,`;
`])),function(t){return t.alignItems},function(t){return t.inline?"inline-flex":"flex"},function(t){return t.direction},function(t){return t.wrap},function(t){var c=t.gap,S=t.theme;return Object(b.a)("gap",c,S)},function(t){return t.justifyContent});n.defaultProps=x,n.propTypes=T},function(r,l,e){var o=e(25),a=e(26),f=e(22),u=e(27);r.exports=function(d,p){return o(d)||a(d,p)||f(d,p)||u()},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l,e){"use strict";e.d(l,"a",function(){return o});var o={DOWN:"ArrowDown",UP:"ArrowUp",RIGHT:"ArrowRight",LEFT:"ArrowLeft",ESCAPE:"Escape",ENTER:"Enter",SPACE:" ",TAB:"Tab",END:"End",HOME:"Home",DELETE:"Delete",PAGE_UP:"PageUp",PAGE_DOWN:"PageDown",BACKSPACE:"Backspace",CLEAR:"Clear"}},,function(r,l){function e(o){return typeof Symbol=="function"&&typeof Symbol.iterator=="symbol"?(r.exports=e=function(a){return typeof a},r.exports.default=r.exports,r.exports.__esModule=!0):(r.exports=e=function(a){return a&&typeof Symbol=="function"&&a.constructor===Symbol&&a!==Symbol.prototype?"symbol":typeof a},r.exports.default=r.exports,r.exports.__esModule=!0),e(o)}r.exports=e,r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l,e){"use strict";e.d(l,"a",function(){return f});var o=e(1),a=0,f=function(u,d){return Object(o.useRef)(d||"".concat(u,"-").concat(++a)).current}},,function(r,l,e){"use strict";e.d(l,"b",function(){return o}),e.d(l,"c",function(){return a}),e.d(l,"a",function(){return f});var o=function(u){return function(d){var p=d.theme,b=d.size;return p.sizes[u][b]}},a=function(){var u=arguments.length>0&&arguments[0]!==void 0?arguments[0]:"&";return function(d){var p=d.theme,b=d.hasError;return`
      outline: none;
      box-shadow: 0;
      transition-property: border-color, box-shadow, fill;
      transition-duration: 0.2s;

      `.concat(u,`:focus-within {
        border: 1px solid `).concat(b?p.colors.danger600:p.colors.primary600,`;
        box-shadow: `).concat(b?p.colors.danger600:p.colors.primary600,` 0px 0px 0px 2px;
      }
    `)}},f=function(u){var d=u.theme;return`
  position: relative;
  outline: none;
  
  &:after {
    transition-property: all;
    transition-duration: 0.2s;
    border-radius: 8px;
    content: '';
    position: absolute;
    top: -4px;
    bottom: -4px;
    left: -4px;
    right: -4px;
    border: 2px solid transparent;
  }

  &:focus-visible {
    outline: none;
    &:after {
      border-radius: 8px;
      content: '';
      position: absolute;
      top: -5px;
      bottom: -5px;
      left: -5px;
      right: -5px;
      border: 2px solid `.concat(d.colors.primary600,`;
    }
  }
`)}},,,function(r,l,e){"use strict";var o=e(20);function a(){}function f(){}f.resetWarningCache=a,r.exports=function(){function u(b,g,O,i,h,v){if(v!==o){var x=new Error("Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types");throw x.name="Invariant Violation",x}}function d(){return u}u.isRequired=u;var p={array:u,bool:u,func:u,number:u,object:u,string:u,symbol:u,any:u,arrayOf:d,element:u,elementType:u,instanceOf:d,node:u,objectOf:d,oneOf:d,oneOfType:d,shape:d,exact:d,checkPropTypes:f,resetWarningCache:a};return p.PropTypes=p,p}},function(r,l,e){"use strict";r.exports="SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED"},function(r,l){r.exports=function(e,o){(o==null||o>e.length)&&(o=e.length);for(var a=0,f=new Array(o);a<o;a++)f[a]=e[a];return f},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l,e){var o=e(21);r.exports=function(a,f){if(a){if(typeof a=="string")return o(a,f);var u=Object.prototype.toString.call(a).slice(8,-1);return u==="Object"&&a.constructor&&(u=a.constructor.name),u==="Map"||u==="Set"?Array.from(a):u==="Arguments"||/^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(u)?o(a,f):void 0}},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l,e){"use strict";e.r(l),e.d(l,"Stack",function(){return C});var o,a,f=e(5),u=e.n(f),d=e(4),p=e.n(d),b=e(3),g=e.n(b),O=e(1),i=e.n(O),h=e(0),v=e.n(h),x=e(2),T=e.n(x),w=e(9),n=["horizontal","spacing","size"],t={size:!0},c=T()(w.Flex).withConfig({shouldForwardProp:function(m,z){return!t[m]&&z(m)}})(o||(o=g()([`
  & > * {
    margin-top: 0;
    margin-bottom: 0;
  }

  & > * + * {
    margin-top: `,`;
  }
`])),function(m){var z=m.theme,P=m.spacing;return z.spaces[P]}),S=T()(w.Flex).withConfig({shouldForwardProp:function(m,z){return!t[m]&&z(m)}})(a||(a=g()([`
  & > * {
    margin-left: 0;
    margin-right: 0;
  }

  & > * + * {
    margin-left: `,`;
  }
`])),function(m){var z=m.theme,P=m.spacing;return z.spaces[P]}),C=Object(O.forwardRef)(function(m,z){var P=m.horizontal,H=m.spacing,M=m.size,I=p()(m,n);return M&&V.warn('Deprecation warning: Usage of "size" prop in Stack component is deprecated. This is discouraged and will be removed in the next major release. Please use "spacing" instead'),P?i.a.createElement(S,u()({ref:z,spacing:H||M},I)):i.a.createElement(c,u()({direction:"column",alignItems:"stretch",ref:z,spacing:H||M},I))});C.displayName="Stack",C.defaultProps={horizontal:!1,size:void 0,spacing:void 0},C.propTypes={horizontal:v.a.bool,size:v.a.number,spacing:v.a.number}},function(r,l){r.exports=function(e,o){if(e==null)return{};var a,f,u={},d=Object.keys(e);for(f=0;f<d.length;f++)a=d[f],o.indexOf(a)>=0||(u[a]=e[a]);return u},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){r.exports=function(e){if(Array.isArray(e))return e},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){r.exports=function(e,o){var a=e==null?null:typeof Symbol!="undefined"&&e[Symbol.iterator]||e["@@iterator"];if(a!=null){var f,u,d=[],p=!0,b=!1;try{for(a=a.call(e);!(p=(f=a.next()).done)&&(d.push(f.value),!o||d.length!==o);p=!0);}catch(g){b=!0,u=g}finally{try{p||a.return==null||a.return()}finally{if(b)throw u}}return d}},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){r.exports=function(){throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`)},r.exports.default=r.exports,r.exports.__esModule=!0},,function(r,l,e){var o=e(42),a=e(43),f=e(22),u=e(44);r.exports=function(d){return o(d)||a(d)||f(d)||u()},r.exports.default=r.exports,r.exports.__esModule=!0},,function(r,l){r.exports=function(e,o,a){return o in e?Object.defineProperty(e,o,{value:a,enumerable:!0,configurable:!0,writable:!0}):e[o]=a,e},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){r.exports=ne},,,,,,,,,function(r,l,e){"use strict";e.r(l),e.d(l,"Icon",function(){return x});var o,a=e(5),f=e.n(a),u=e(3),d=e.n(u),p=e(1),b=e.n(p),g=e(0),O=e.n(g),i=e(6),h=e(2),v=e.n(h)()(i.Box)(o||(o=d()([`
  path {
    fill: `,`;
  }
  `,`
`])),function(T){var w=T.color;return T.theme.colors[w]},function(T){var w=T.theme;return(0,T.colors)(w)}),x=b.a.forwardRef(function(T,w){return b.a.createElement(v,f()({ref:w},T))});x.displayName="Icon",x.defaultProps={color:"neutral600",colors:function(){}},x.propTypes={color:O.a.string,colors:O.a.func}},function(r,l,e){var o=e(21);r.exports=function(a){if(Array.isArray(a))return o(a)},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){r.exports=function(e){if(typeof Symbol!="undefined"&&e[Symbol.iterator]!=null||e["@@iterator"]!=null)return Array.from(e)},r.exports.default=r.exports,r.exports.__esModule=!0},function(r,l){r.exports=function(){throw new TypeError(`Invalid attempt to spread non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`)},r.exports.default=r.exports,r.exports.__esModule=!0},,,,,,,,,,,,,,,,,,,,function(r,l,e){"use strict";e.r(l),e.d(l,"TextButton",function(){return C});var o,a=e(5),f=e.n(a),u=e(4),d=e.n(u),p=e(3),b=e.n(p),g=e(1),O=e.n(g),i=e(0),h=e.n(i),v=e(2),x=e.n(v),T=e(6),w=e(8),n=e(9),t=e(16),c=["children","startIcon","endIcon","onClick","disabled"],S=x()(n.Flex)(o||(o=b()([`
  background: transparent;
  border: none;

  &[aria-disabled='true'] {
    pointer-events: none;
    svg path {
      fill: `,`;
    }
  }

  svg {
    display: flex;
    font-size: `,`rem;
  }

  svg path {
    fill: `,`;
  }

  `,`
`])),function(m){return m.theme.colors.neutral600},.625,function(m){return m.theme.colors.primary600},t.a),C=O.a.forwardRef(function(m,z){var P=m.children,H=m.startIcon,M=m.endIcon,I=m.onClick,D=m.disabled,U=d()(m,c),F=I&&!D?I:void 0;return O.a.createElement(S,f()({ref:z,"aria-disabled":D,onClick:F,as:"button",type:"button"},U),H&&O.a.createElement(T.Box,{as:"span",paddingRight:2,"aria-hidden":!0},H),O.a.createElement(w.Typography,{variant:"pi",textColor:D?"neutral600":"primary600"},P),M&&O.a.createElement(T.Box,{as:"span",paddingLeft:2,"aria-hidden":!0},M))});C.displayName="TextButton",C.defaultProps={disabled:!1,startIcon:void 0,endIcon:void 0,onClick:void 0},C.propTypes={children:h.a.node.isRequired,disabled:h.a.bool,endIcon:h.a.element,onClick:h.a.func,startIcon:h.a.element}},,,,,,,,,,,,,,,,function(r,l,e){"use strict";e.r(l),e.d(l,"KeyboardNavigable",function(){return T});var o=e(5),a=e.n(o),f=e(29),u=e.n(f),d=e(4),p=e.n(d),b=e(1),g=e.n(b),O=e(0),i=e.n(O),h=e(6),v=e(11),x=["tagName","attributeName"],T=function(w){var n=w.tagName,t=w.attributeName,c=p()(w,x),S=function(){var m=document.activeElement;return n?m.tagName.toLowerCase()===n:m.hasAttribute(t)},C=function(m){return n?m.querySelectorAll(n):m.querySelectorAll("[".concat(t,"]"))};return g.a.createElement(h.Box,a()({onKeyDown:function(m){switch(m.key){case v.a.RIGHT:case v.a.DOWN:m.preventDefault();var z=document.activeElement;if(S()){m.preventDefault();var P=u()(C(m.currentTarget)),H=P.findIndex(function(F){return F===z});P[H+1<P.length?H+1:0].focus()}break;case v.a.LEFT:case v.a.UP:m.preventDefault();var M=document.activeElement;if(S()){m.preventDefault();var I=u()(C(m.currentTarget)),D=I.findIndex(function(F){return F===M});I[D-1>-1?D-1:I.length-1].focus()}break;case v.a.HOME:S()&&(m.preventDefault(),C(m.currentTarget).item(0).focus());break;case v.a.END:if(S()){m.preventDefault();var U=C(m.currentTarget);U.item(U.length-1).focus()}}}},c))};T.defaultProps={attributeName:void 0,tagName:void 0},T.propTypes={attributeName:i.a.string,tagName:i.a.string}},,,,,,,,,,,,,,,,,,,function(r,l,e){"use strict";e.r(l),e.d(l,"AccordionTypography",function(){return S}),e.d(l,"Accordion",function(){return m}),e.d(l,"AccordionContent",function(){return D}),e.d(l,"AccordionToggle",function(){return re}),e.d(l,"AccordionGroup",function(){return te});var o,a,f=e(31),u=e.n(f),d=e(3),p=e.n(d),b=e(1),g=e.n(b),O=e(0),i=e.n(O),h=e(2),v=e.n(h),x=e(8),T=Object(b.createContext)(),w=function(){return Object(b.useContext)(T)},n=e(14),t=e(6),c=e(9),S=v()(x.Typography)(o||(o=p()([""]))),C=v()(t.Box)(a||(a=p()([`
  border: `,`;

  &:hover:not([aria-disabled='true']) {
    border: 1px solid `,`;

    `,` {
      color: `,`;
    }

    `,` {
      color: `,`;
    }

    & > `,` {
      background: `,`;
    }

    [data-strapi-dropdown='true'] {
      background: `,`;
    }
  }
`])),function(s){var y=s.theme,E=s.expanded,j=s.variant,N=s.disabled;return s.error?"1px solid ".concat(y.colors.danger600," !important"):"1px solid ".concat(N?y.colors.neutral150:E?y.colors.primary600:j==="primary"?y.colors.neutral0:y.colors.neutral100)},function(s){return s.theme.colors.primary600},S,function(s){var y=s.theme;return s.expanded?void 0:y.colors.primary700},x.Typography,function(s){var y=s.theme;return s.expanded?void 0:y.colors.primary600},c.Flex,function(s){return s.theme.colors.primary100},function(s){return s.theme.colors.primary200}),m=function(s){var y=s.children,E=s.expanded,j=s.id,N=s.size,_=s.variant,W=s.disabled,k=s.error,B=s.hasErrorMessage,$=s.onToggle,J=s.toggle,R=Object(n.a)("accordion",j);return g.a.createElement(T.Provider,{value:{expanded:E,onToggle:$,toggle:J,id:R,size:N,variant:_,disabled:W}},g.a.createElement(C,{"data-strapi-expanded":E,disabled:W,"aria-disabled":W,expanded:E,hasRadius:!0,variant:_,error:k},y),k&&B&&g.a.createElement(t.Box,{paddingTop:1},g.a.createElement(x.Typography,{variant:"pi",textColor:"danger600"},k)))};m.defaultProps=u()({disabled:!1,error:void 0,expanded:!1,hasErrorMessage:!0,id:void 0,toggle:!1,size:"M",variant:"primary",onToggle:void 0},"toggle",void 0),m.propTypes={children:i.a.node.isRequired,disabled:i.a.bool,error:i.a.string,expanded:i.a.bool,hasErrorMessage:i.a.bool,id:i.a.string,onToggle:i.a.func,size:i.a.oneOf(["S","M"]),toggle:i.a.func,variant:i.a.oneOf(["primary","secondary"])};var z=e(5),P=e.n(z),H=e(4),M=e.n(H),I=["children"],D=function(s){var y=s.children,E=M()(s,I),j=w(),N=j.expanded,_=j.id;if(!N)return null;var W="accordion-content-".concat(_),k="accordion-label-".concat(_),B="accordion-desc-".concat(_);return g.a.createElement(t.Box,P()({role:"region",id:W,"aria-labelledby":k,"aria-describedby":B},E),y)};D.propTypes={children:i.a.node.isRequired};var U,F,xe=e(32),ye=e.n(xe),Te=e(64),ie=e(23),we=e(41),Oe=["title","description","as","togglePosition","action"],se=v()(Te.TextButton)(U||(U=p()([`
  text-align: left;

  svg {
    width: `,`rem;
    height: `,`rem;

    path {
      fill: `,`;
    }
  }
`])),.875,.875,function(s){var y=s.theme;return s.expanded?y.colors.primary600:y.colors.neutral500}),ce=v()(c.Flex)(F||(F=p()([`
  height: `,`;
  border-radius: `,`;

  &:hover {
    svg {
      path {
        fill: `,`;
      }
    }
  }
`])),function(s){var y=s.theme,E=s.size;return y.sizes.accordions[E]},function(s){var y=s.theme;return s.expanded?"".concat(y.borderRadius," ").concat(y.borderRadius," 0 0"):y.borderRadius},function(s){return s.theme.colors.primary600}),re=function(s){var y=s.title,E=s.description,j=s.as,N=s.togglePosition,_=s.action,W=M()(s,Oe),k=Object(b.useRef)(null),B=w(),$=B.onToggle,J=B.toggle,R=B.expanded,oe=B.id,G=B.size,Pe=B.variant,L=B.disabled,pe="accordion-content-".concat(oe),Q="accordion-label-".concat(oe),fe="accordion-desc-".concat(oe),Z=G==="M"?6:4,ge=function(K){var Re=K.expanded,_e=K.disabled,ke=K.variant;return Re?"primary100":_e?"neutral150":ke==="primary"?"neutral0":"neutral100"}({expanded:R,disabled:L,variant:Pe}),ee=R?"primary600":"neutral700",me=R?"primary600":"neutral600",je=R?"primary200":"neutral200",he="".concat(G==="M"?2:1.5,"rem"),be=function(){L||(J&&!$?(V.warn('Deprecation warning: Usage of "toggle" prop in Accordion component is deprecated. This is discouraged and will be removed in the next major release. Please use "onToggle" instead'),J()):$())},ve=g.a.createElement(c.Flex,{justifyContent:"center",borderRadius:"50%",height:he,width:he,transform:R?"rotate(180deg)":void 0,"data-strapi-dropdown":!0,disabled:L,"aria-hidden":!0,as:"span",background:je,cursor:L?"not-allowed":"pointer",onClick:function(){var K;return k==null||(K=k.current)===null||K===void 0?void 0:K.click()}},g.a.createElement(we.Icon,{as:ye.a,width:"".concat(G==="M"?11/16:.5,"rem"),color:R?"primary600":"neutral600"}));return N==="left"?g.a.createElement(ce,{paddingLeft:Z,paddingRight:Z,background:ge,expanded:R,justifyContent:"space-between",size:G,cursor:L?"not-allowed":""},g.a.createElement(ie.Stack,{horizontal:!0,spacing:3,flex:1},ve,g.a.createElement(se,P()({ref:k,onClick:be,"aria-disabled":L,"aria-expanded":R,"aria-controls":pe,"aria-labelledby":Q,"data-strapi-accordion-toggle":!0,expanded:R,type:"button",flex:1},W),g.a.createElement(g.a.Fragment,null,G==="S"?g.a.createElement(x.Typography,{fontWeight:"bold",as:j,id:Q,textColor:ee},y):g.a.createElement(S,{variant:"delta",as:j,id:Q,textColor:ee},y),E&&g.a.createElement(x.Typography,{as:"p",id:fe,textColor:me},E)))),_):g.a.createElement(ce,{paddingRight:Z,paddingLeft:Z,background:ge,expanded:R,size:G,justifyContent:"space-between",cursor:L?"not-allowed":""},g.a.createElement(se,P()({ref:k,onClick:be,"aria-disabled":L,"aria-expanded":R,"aria-controls":pe,"aria-labelledby":Q,"data-strapi-accordion-toggle":!0,expanded:R,type:"button",flex:1},W),g.a.createElement(g.a.Fragment,null,G==="S"?g.a.createElement(x.Typography,{fontWeight:"bold",as:j,id:Q,textColor:ee},y):g.a.createElement(x.Typography,{variant:"delta",as:j,id:Q,textColor:ee},y),E&&g.a.createElement(x.Typography,{as:"p",id:fe,textColor:me},E))),g.a.createElement(ie.Stack,{horizontal:!0,spacing:3},ve,_))};re.defaultProps={action:void 0,as:"span",description:void 0,variant:"primary",togglePosition:"right"},re.propTypes={action:i.a.node,as:i.a.string,description:i.a.string,title:i.a.string.isRequired,togglePosition:i.a.oneOf(["right","left"]),variant:i.a.oneOf(["primary","secondary"])};var ue,le,de,Ee=e(80),Se=v()(t.Box)(ue||(ue=p()([`
  border-bottom: 1px solid `,`;
  border-right: 1px solid `,`;
  border-left: 1px solid `,`;
  border-radius: 0 0 `," ",`;
`])),function(s){return s.theme.colors.neutral200},function(s){return s.theme.colors.neutral200},function(s){return s.theme.colors.neutral200},function(s){return s.theme.borderRadius},function(s){return s.theme.borderRadius}),Ce=v()(t.Box)(le||(le=p()([`
  & > * {
    & > * {
      border-radius: unset;
    }
  }

  & > * {
    border-radius: unset;
    border-right: 1px solid `,`;
    border-left: 1px solid `,`;
    border-bottom: 1px solid `,`;
  }

  & > *:first-of-type {
    border-top: 1px solid `,`;
    border-radius: `," ",` 0 0;
    & > * {
      border-radius: `," ",` 0 0;
    }

    &:hover {
      border-top: 1px solid `,`;
    }
  }

  & [data-strapi-expanded='true'] {
    border: 1px solid `,`;
  }

  `,`
`])),function(s){return s.theme.colors.neutral200},function(s){return s.theme.colors.neutral200},function(s){return s.theme.colors.neutral200},function(s){return s.theme.colors.neutral200},function(s){return s.theme.borderRadius},function(s){return s.theme.borderRadius},function(s){return s.theme.borderRadius},function(s){return s.theme.borderRadius},function(s){return s.theme.colors.primary600},function(s){return s.theme.colors.primary600},function(s){var y=s.theme,E=s.footer;return`
    &:not(`.concat(E,`) {
      & > *:last-of-type {
        border-radius: 0 0 `).concat(y.borderRadius," ").concat(y.borderRadius,`;
      }
    }
  `)}),ze=v()(t.Box)(de||(de=p()([`
  svg path {
    fill: `,`;
  }
`])),function(s){return s.theme.colors.neutral500}),te=function(s){var y=s.children,E=s.footer,j=s.label,N=s.labelAction,_=s.error,W=b.Children.toArray(y).map(function(k){return Object(b.cloneElement)(k,{hasErrorMessage:!1})});return g.a.createElement(Ee.KeyboardNavigable,{attributeName:"data-strapi-accordion-toggle"},j&&g.a.createElement(c.Flex,{paddingBottom:1},g.a.createElement(x.Typography,{variant:"pi",as:"label",textColor:"neutral800",fontWeight:"bold"},j),N&&g.a.createElement(ze,{paddingLeft:1},N)),g.a.createElement(Ce,{footer:E},W),E&&g.a.createElement(Se,null,E),_&&g.a.createElement(t.Box,{paddingTop:1},g.a.createElement(x.Typography,{variant:"pi",textColor:"danger600"},_)))};te.defaultProps={footer:null,error:void 0,label:null,labelAction:void 0},te.propTypes={children:i.a.node.isRequired,error:i.a.string,footer:i.a.node,label:i.a.string,labelAction:i.a.node}}])})},37213:(X,ae,A)=>{var V=A(26460),Y=A(88904),q=A(52920),ne=A(77236);function r(l,e){var o=ne(l)?V:q;return o(l,Y(e,3))}X.exports=r}}]);
